text-mode-core/tsconfig.json

```diff
@@ -2,13 +2,12 @@
   "extends": "../tsconfig-base",
   "compilerOptions": {},
   "references": [
+    // https://github.com/microsoft/TypeScript/issues/33685
+    // { "path": "../src" },
     { "path": "../localization" },
     { "path": "../metadata" },
-    { "path": "../parsing" }
+    { "path": "../parsing" },
+    { "path": "../graph-state" }
   ],
-  "include": [
-    "**/*",
-    "../src/tests/run_calc_for_tests.ts",
-    "../loaders/utils.mjs"
-  ]
+  "include": ["**/*", "../src", "../loaders/utils.mjs"]
```

Comment:
Checking my understanding: could you confirm if this comment is right?

The current project structure is unfortunate here --
- `src` references `text-mode-core` of course to do text-mode stuff.
- `text-mode-core` references `src` just for `src/tests/run_calc_for_tests`.
- Entirely reasonably, cyclic references are not allowed (https://github.com/microsoft/TypeScript/issues/33685 as you note), so the easiest way to make this work is to say that `src/tests/run_calc_for_tests` is part of both projects.

Maybe it would be better to split off `src/tests` into its own project. But that's out of the scope of this PR.